src/common/charts/shared.js

```diff
@@ -0,0 +1,99 @@
+    trendPositiveColor = 'var(--color-data-viz-trend-positive)',
+    trendNegativeColor = 'var(--color-data-viz-trend-negative)',
+    chartPrimaryColor = 'var(--color-data-viz-chart-primary)',
+    chartSecondaryColor = 'var(--color-data-viz-chart-secondary)',
+    chartTertiaryBackgroundColor = 'var(--color-data-viz-chart-tertiary-background)',
+    chartTertiaryStrokeColor = 'var(--color-data-viz-chart-tertiary-stroke)',
+    chartQuaternaryBackgroundColor = 'var(--color-data-viz-chart-quaternary-background)',
+    chartQuaternaryStrokeColor = 'var(--color-data-viz-chart-quaternary-stroke)',
+    chartQuinaryBackgroundColor = 'var(--color-data-viz-chart-quinary-background)',
+    chartQuinaryStrokeColor = 'var(--color-data-viz-chart-quinary-stroke)',
+    // patterns are in highcharts PatternOptionsObject format
+    // refer to https://api.highcharts.com/class-reference/Highcharts.PatternOptionsObject
+    pattern1 = {
+        pattern: {
+            path: {
+                // d is a standard SVG path definition string
+                // refer to https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d
+                d: 'M0 0 L0 3', // draw a 3 until vertical line
+            },
+            width: 4.5, // defines the x bounds of the repeating pattern
+            height: 3, // defines the  y bounds of the repeating pattern
+            backgroundColor: chartTertiaryBackgroundColor, // sets the patterns background color
+            color: chartTertiaryStrokeColor, // sets the patterns stroke color
+            patternTransform: 'rotate(-60)', // rotates the path -60 degrees
+        },
+    },
+    pattern2 = {
+        pattern: {
+            path: {
+                // d is a standard SVG path definition string
+                // refer to https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d
+                d: 'M0 0 L3 0',
+            },
+            width: 3, // defines the x bounds of the repeating pattern
+            height: 5, // defines the  y bounds of the repeating pattern
+            backgroundColor: chartQuaternaryBackgroundColor, // sets the patterns background color
+            color: chartQuaternaryStrokeColor, // sets the patterns stroke color
+        },
+    },
+    // configure the color sequences based on how many series are used
+    colorsSingle = [chartPrimaryColor],
+    colorsTwo = [chartSecondaryColor, chartPrimaryColor],
+    colorsThree = [chartSecondaryColor, chartPrimaryColor, pattern1],
+    colorsMore = [
+        chartSecondaryColor,
+        pattern1,
+        chartPrimaryColor,
+        pattern2,
+        chartQuinaryBackgroundColor,
+    ],
+    // function is used to set up the colors including lineColor(svg stroke) on each of the series objects
+    // based on the length of the series array
+    setSeriesColors = function (series) {
+        let colors;
+        switch (series.length) {
+            case 1:
+                colors = colorsSingle;
```

Comment:
@agliga , I just realized this function is just cherry-picking colors outside of itself. This is a bit brittle. We can keep it like this for now if we just want to get it in, but we should refactor this. Ideally, if we use a two-dimensional array here and pass the array into the function call, the function can then loop through and set the proper series colors without conditionals.